reporterdashboard/dashboard.py

```diff
@@ -0,0 +1,112 @@
+import json
+import argparse
+
+
+def get_matching_test_cases(test_cases_1, test_cases_2):
+    """Returns test cases that are both in test_cases_1 and test_cases_2.
+
+    Test cases are ordered by their name.
+    """
+    # create sorted copies
+    sorted_1 = sorted(test_cases_1, key=lambda d: d["name"])
+    sorted_2 = sorted(test_cases_2, key=lambda d: d["name"])
+    overlapping_test_cases = []
+
+    while True:
+        if len(sorted_1) == 0 or len(sorted_2) == 0:
+            break
+
+        if sorted_1[0]["name"] == sorted_2[0]["name"]:
+            # only include passing tests
+            if sorted_1[0]["result"] != "pass" or sorted_2[0]["result"] != "pass":
+                sorted_1.pop(0)
+                sorted_2.pop(0)
+                continue
+            overlapping_test_cases.append((sorted_1.pop(0), sorted_2.pop(0)))
+        elif sorted_1[0]["name"] < sorted_2[0]["name"]:
+            sorted_1.pop(0)
+        else:
+            sorted_2.pop(0)
+    return overlapping_test_cases
+
+
+def print_energy_differences(data):
+    """Prints the energy difference for every test combination tuple in data.
+
+    Keyword arguments:
+    data -- list of tuples each containing two test measurements
+    """
+    for testcombi in data:
+        # calculate average information
+        if len(testcombi) != 2:
+            raise TypeError(
+                f"Data tuple should have two data objects (actual is {len(testcombi)})."
+            )
+        N_0 = testcombi[0]["N"]
+        e_0 = 0
+        p_0 = 0
+        t_0 = 0
+        for i in range(0, N_0):
+            e_0 += testcombi[0]["energy"][i]
+            p_0 += testcombi[0]["power"][i]
+            t_0 += testcombi[0]["execution_time"][i]
+        e_0 /= N_0
+        p_0 /= N_0
+        t_0 /= N_0
```

Comment:
can't this just be a sum? Or at least make it a utility function, because this is quite verbose and repetitive

Reply:
You are so right. I'll change that